docs/developer/graphql.md

```diff
@@ -236,3 +248,57 @@ import * as graphql from "./typings/types";
 }
 ```
 
+## Example
+
+The following GraphQL
+subscribes to completed builds, returning related data such as the
+last commit and any linked Slack channels:
+
+[graphql]: http://graphql.org (GraphQL)
+
+```graphql
+subscription OnBuildComplete {
+  Build {
+    buildId
+    buildUrl
+    compareUrl
+    name
+    status
+    commit {
+      sha
+      message
+      repo {
+        name
+        owner
+        gitHubId
+        allowRebaseMerge
+        channels {
+          name
+          id
+        }
+      }
+      statuses {
+        context
+        description
+        state
+        targetUrl
+      }
+    }
+  }
+}
+```
+
+When using TypeScript (our recommended language), an event handler can
+subscribe to such events with the benefit of strong typing. For
+example, this Atomist event handler can respond to the above GraphQL
+subscription:
+
+```typescript
+@EventHandler("Set status on build complete",
+    GraphQL.subscriptionFromFile("graphql/subscription/OnBuildComplete.graphql"))
+export class SetStatusOnBuildComplete implements HandleEvent<OnBuildComplete.Subscription> {
+
+    public async handle(event: EventFired<OnBuildComplete.Subscription>,
+    	ctx: HandlerContext,
+    	params: this): Promise<HandlerResult> {
+```
```

Comment:
ooh. Not in this PR, but we need to document how those types get generated. Is that anywhere in the docs currently, @gjtorikian  ? (like have you found it? I don't think it is. I know what to write though)